akka-javasdk/src/main/java/akka/javasdk/agent/ConversationMemory.java

```diff
@@ -0,0 +1,171 @@
+      messages = messages != null ? new LinkedList<>(messages) : new LinkedList<>();
+      enforceMaxCapacity(messages, maxSize);
+    }
+
+    public boolean isEmpty() {
+      return messages.isEmpty();
+    }
+
+    public State withMaxSize(int newMaxSize) {
+      List<ConversationMessage> updatedMessages = new LinkedList<>(messages);
+      return new State(newMaxSize, updatedMessages);
+    }
+
+    public State addMessage(ConversationMessage message) {
+      List<ConversationMessage> updatedMessages = new LinkedList<>(messages);
+      updatedMessages.add(message);
+
+      return new State(maxSize, updatedMessages);
+    }
+
+    public State clear() {
+      return new State(maxSize, new LinkedList<>());
+    }
+
+    private static void enforceMaxCapacity(List<ConversationMessage> messages, int maxSize) {
+
+      // If we exceed the maximum size, remove the oldest message
+      while (messages.size() > maxSize) {
+        logger.debug("Erasing oldest message from history, remaining={}, maxSize={}", messages.size() - 1, maxSize);
+        messages.removeFirst();
+      }
+    }
+
+  }
+  
+  @Override
+  public State emptyState() {
+    // FIXME: load default from config?
+    return new State(1000, new LinkedList<>());
+  }
+
+  /**
+   * Sealed interface representing events that can occur in the ConversationMemory entity.
+   */
+  public sealed interface Event {
+
+    @TypeName("akka-memory-limited-window-set")
+    record LimitedWindowSet(int maxSize) implements Event {
+    }
+
+    @TypeName("akka-memory-user-message-added")
+    record UserMessageAdded(String message) implements Event {
+    }
+
+    @TypeName("akka-memory-ai-message-added")
+    record AiMessageAdded(String message) implements Event {
+    }
+    
+    @TypeName("akka-memory-deleted")
+    record Deleted() implements Event {
+    }
+  }
+
+  // Request commands
+  public record LimitedWindow(int maxSize) {}
+
+  public Effect<Done> setLimitedWindow(LimitedWindow limitedWindow) {
+    if (limitedWindow.maxSize <= 0) {
+      return effects().error("Maximum size must be greater than 0");
+    } else {
+      return effects()
+          .persist(new Event.LimitedWindowSet(limitedWindow.maxSize))
+          .thenReply(__ -> done());
+    }
+  }
+
+  public Effect<Done> addUserMessage(String message) {
+    log.debug("Adding user message: {}", message);
+    return effects()
+        .persist(new Event.UserMessageAdded(message))
+        .thenReply(__ -> Done.done());
+  }
+
+  public Effect<Done> addAiMessage(String message) {
+    return effects()
+        .persist(new Event.AiMessageAdded(message))
+        .thenReply(__ -> Done.done());
+  }
+
+  public record AddInteractionCmd(String userMessage, String aiMessage) {
+
+  }
+  public Effect<Done> addInteraction(AddInteractionCmd cmd) {
+    log.debug("Adding interaction: user={}, ai={}", cmd.userMessage, cmd.aiMessage);
```

Comment:
We must not log the content of user and ai messages, since those may contain sensitive information that shouldn't end up in logging infra.

Reply:
good catch, tks